packages/connect/src/device/Device.ts

```diff
@@ -823,6 +860,8 @@ export class Device extends TypedEmitter<DeviceEvents> {
         const enabled = DataManager.getSettings('enableFirmwareHashCheck');
         if (!enabled || baseUrl === undefined) return createFailResult('check-skipped');
 
+        const localFirmwares = DataManager.getLocalFirmwares();
```

Comment:
I am implementing it now, but the idea is to have an interface that let us use the local firmware when they are available. So this function will maybe take model and version we need and response if it is suppose to be locally downloaded. And if so, use it.

Reply:
Right now the information I have from desktop is something like:

```
{
  userDataDir: '/home/user/.config/@trezor/suite-desktop-local',
  firmwares: [
    'trezor-t2t1-2.8.9-regular.bin',
    'trezor-t3b1-2.8.9-bitcoinonly.bin'
  ]
```
Which should be enough to be able use one of those firmware if we need them.